brainiak/utils/simulator.py

```diff
@@ -0,0 +1,1058 @@
+
+    # Scale the mask according to the input brain
+    # You might get a warning but ignore it
+    mask = ndimage.zoom(mask_raw, zoom_factor, order=2)
+
+    # Anything that is in the bottom 1% (presumably the exterior of the
+    # brain, make zero)
+    mask[mask < 0.01] = 0
+
+    # Mask out the non grey matter regions
+    brain = volume * mask
+
+    return brain
+
+
+def plot_brain(brain,
+               percentile=99,
+               starting_tr=0):
+    """ Display the brain that has been generated with a given threshold
+
+    Parameters
+    ----------
+
+    brain : multidimensional array
+        This is a 3d or 4d array with the neural data
+
+    percentile : float
+        What percentage of voxels will be included? Based on the values
+        supplied
+
+    starting_tr : int
+        What is the TR that this starts at
+
+
+    """
+
+    import matplotlib.pyplot as plt
+    # This is necessary to make add_subplot work
+    from mpl_toolkits.mplot3d import Axes3D as temp
+
+    temp.__module__  # Do in order to remove the pep8 error generated above
+
+    fig = plt.figure()
+    ax = fig.add_subplot(111, projection='3d')
+
+    # If there is only one volume then add a fourth dimension to it
+    if len(brain.shape) == 3:
+        temp = np.zeros((brain.shape[0], brain.shape[1], brain.shape[2],
+                         1))  # Make a new matrix
+        temp[:, :, :, 0] = brain  # Insert the brain here
+        brain = temp  # rename
+
+    # Threshold the data
+
+    brain_threshold_old = []
+    for tr_counter in list(range(starting_tr, brain.shape[3])):
+
+        Threshold = np.percentile(
+            brain[:, :, :, tr_counter].reshape(np.prod(brain.shape[0:3])),
+            percentile)
+
+        # How many voxels exceed a threshold
+        brain_threshold = np.where(
+            np.abs(brain[:, :, :, tr_counter]) > Threshold)
+
+        brain_all = np.where(np.abs(brain[:, :, :, tr_counter]) > 0)
+
+        # Clear the way
+        ax.clear()
+
+        ax.set_xlim(0, brain.shape[0])
+        ax.set_ylim(0, brain.shape[1])
+        ax.set_zlim(0, brain.shape[2])
+
+        ax.scatter(brain_all[0],
+                   brain_all[1],
+                   brain_all[2],
+                   zdir='z',
+                   c='black',
+                   s=10,
+                   alpha=0.01)
+
+        ax.scatter(brain_threshold[0],
+                   brain_threshold[1],
+                   brain_threshold[2],
+                   zdir='z',
+                   c='red',
+                   s=20)
+
+        if tr_counter > 0:
+            ax.scatter(brain_threshold_old[0],
+                       brain_threshold_old[1],
+                       brain_threshold_old[2],
+                       zdir='z',
+                       c='red',
+                       s=10)
+
+        # Store the most recent volume
+        brain_threshold_old = brain_threshold
+
+        # Wait for an input
+        print(tr_counter)
+        plt.pause(0.5)
```

Comment:
BrainIAK code must not do I/O, therefore we use logging instead of printing to standard output. You must not interact with the GUI by calling `plt.pause`. I suggest you return the figure for the calling code to display or save.